editor/src/components/editor/actions/actions.tsx

```diff
@@ -1914,6 +1915,25 @@ export const UPDATE_FNS = {
       toasts: uniqToasts([...withOldToastRemoved.toasts, action.toast]),
     }
   },
+  UPDATE_GITHUB_OPERATIONS: (
+    action: UpdateGithubOperations,
+    editor: EditorModel,
+    _dispatch: EditorDispatch,
+  ): EditorModel => {
+    const operations = [...editor.githubOperations]
+    if (action.type === 'add') {
```

Comment:
We tend to use switch statements that we make exhaustive by the use of this trick with the `never` type: https://github.com/concrete-utopia/utopia/blob/a24e4baffbbd02e996f34b0e77eaeea80b9b1fa5/editor/src/components/editor/actions/actions.tsx#L1669-L1671
So that if a new operation type is added then we'll get a compiler error from tsc to indicate we've missed something out.

Reply:
alright, this is very cool. https://github.com/concrete-utopia/utopia/pull/2707/commits/7d50b367a37186254a2575f25816de9e5fff1f77